platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/module/telemetry/PerformanceEvent.java

```diff
@@ -1,46 +1,109 @@
 package com.mapbox.mapboxsdk.module.telemetry;
 
+import com.google.gson.Gson;
+
+import com.google.gson.JsonObject;
+
+import com.google.gson.reflect.TypeToken;
 import com.mapbox.android.telemetry.Event;
 
 import android.os.Bundle;
 import android.os.Parcel;
 
+
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
+import java.util.List;
 import java.util.Locale;
 
 /**
  * Generic Performance Event that can be used for performance measurements.
  * Customer measurements can be added to the bundle.
  */
 public class PerformanceEvent extends Event {
-  private static final String PERFORMANCE_TRACE = "performance.trace";
+
+  private static final String PERFORMANCE_TRACE = "mobile.performance_trace";
 
   private final String event;
 
   private final String created;
 
   private final String sessionId;
 
-  private final Bundle data;
+  private final List<Attribute<String>> attributes;
+
+  private final List<Attribute<Double>> counters;
+
+  private final JsonObject metadata;
+
 
   private static final SimpleDateFormat DATE_FORMAT =
           new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ", Locale.US);
 
-  PerformanceEvent(String sessionId, Bundle data) {
+  PerformanceEvent(String sessionId, Bundle bundle) {
+
     this.event = PERFORMANCE_TRACE;
     this.created = DATE_FORMAT.format(new Date());
     this.sessionId = sessionId;
-    this.data = data;
+
+    Gson gson = new Gson();
+    String strInBundle = bundle.getString("attributes");
+    if (strInBundle == null) {
+      this.attributes = null;
+    } else {
+      this.attributes = gson.fromJson(strInBundle,
+              new TypeToken<ArrayList<Attribute<String>>>() {}.getType());
+    }
+
+    strInBundle = bundle.getString("counters");
+    if (strInBundle == null) {
+      this.counters = null;
+    } else {
+      this.counters = gson.fromJson(strInBundle,
+              new TypeToken<ArrayList<Attribute<? extends Double>>>() {}.getType());
+    }
+
+    strInBundle = bundle.getString("metadata");
+    if (strInBundle == null) {
+      this.metadata = null;
+    } else {
+      this.metadata = gson.fromJson(strInBundle, JsonObject.class);
+    }
```

Comment:
above code block consist out 3 similar looking if blocks, can be refactored into a method instead.